app/controllers/users_controller.rb

```diff
@@ -10,11 +10,15 @@ def update
     @user = User.find(params[:id])
     raise AuthorizationError if current_user != @user
 
-    @user.update_attributes({
-      sam_id:   params[:sam_id],
-      duns_id:  params[:duns_id]
-    })
+    @user.update_attributes(user_params)
```

Comment:
this isn't important but thought I'd share a recent learning: `update_attributes` is actually an alias for `update` http://api.rubyonrails.org/classes/ActiveRecord/Persistence.html#method-i-update

so you could just call `@user.update(user_params)` here